### PR TITLE
Create emme_verkko.md

### PR DESCRIPTION
**Pull request by zptro**
Added EMME network coding instructions.

Also decreased line length for some old files.

### CODE REVIEW

**Comment by zptro** [comment]
Oh, how wonderful it would be to get this released, @johpiip ;)

**Comment by johpiip** [comment]
@zptro I was planning on asking Mervi to review this after she comes back from her vacation, but if you consider it a masterpiece, I can merge it right away. 😄

File: docs/emme_verkko.md [hunk 1]
```diff
@@ -0,0 +1,643 @@
...
+|	vdf	            | sijoittelufunktion nro                                       | ei käytössä (0) | ei käytössä (0)                     |
+| ul1	| linkin yhden kaistan kapasiteetti | raitiovaunuverkko: aamu-, päivä- ja iltaliikenteen nopeus, juna- ja metroverkko: ei käytössä (0) | ei käytössä (0) |
+| ul2	            | linkin vapaa nopeus                                          | ei käytössä (0) | ei käytössä (0)                     |
+| ul3           	| autoliikenteen linkeille sijoitellaan aluksi raskas liikenne, ja tulokset tallennetaan linkkiattribuuttiin ul3 (oltava link user datassa eikä extra-attribuutissa, jotta tietoa voidaan käyttää sijoittelufunktioissa)         | ei käytössä (0) | ei käytössä (0)                     |
+| @pyoratieluokka	|                                                              |                 | pyörätien laatuluokka (taulukko 12) |
+
+### Linkkityypit, väyläluokat ja sallitut kulkutavat
+
+Sijoitteluskripti muuttaa linkien funktio-, nopeus- ja kapasiteettiattribuutit (vdf, ul1, ul2) type-attribuutin perusteella.
```
[the remark this answers is not in the document]
Comment (johpiip) [reply docs/emme_verkko.md:228]: But are new overriden values also written to links, or do the "wrong" attributes stay as they are?

File: docs/emme_verkko.md [hunk 2]
```diff
@@ -0,0 +1,643 @@
...
+Kävely ja pyöräily on lähtökohtaisesti sallittua autolinkeillä ja kiellettyä raideliikenteen linkeillä.
+Ratikka- ja pikaratikkalinkeille voi koodata kävelyn/pyöräilyn (”taf”, ”paf”, ”tpaf”), jos ei ole rinnakkaista ajoneuvolinkkiä, jolle kävelyn voisi koodata.
+Metro- ja junaradoilla kävelyä/pyöräilyä ei kuitenkaan voi sallia (eli ”maf”, ”rjaf” eivät ole sallittuja).
+Autolinkkejä, joilla kävely ja pyöräily eivät ole sallittuja, voi koodata poikkeustapauksissa, kuten tunnelit (esimerkiksi Kampin terminaalissa).
+Käytännössä pyöräily on verkossa sallittu kaikkialla missä kävelykin, joitakin poikkeuksia lukuun ottamatta.
+Muutamissa paikoissa on erikseen estetty pyöräily, jos se ei ole todellisuudessa mahdollista.
+Pyöräily on sallittua moottoritiemäisten väylien linkeillä molempiin suuntiin samoin kuin kävely,
+jos moottoritien rinnalla tai läheisyydessä kulkee kävely- ja pyörätie.
+
+*Taulukko 10. Linkkityypit (link type)*
```
Comment (johpiip) [inline docs/emme_verkko.md:251]: Modes in this table do not include `h`. Should they?

**Comment by zptro** [comment]
@johpiip, she actually already reviewed it (I am not 100 % sure it was a complete review though), but she did not do it in Github, she did it as a Teams message to me.

File: docs/emme_verkko.md [hunk 3]
```diff
@@ -0,0 +1,643 @@
...
+| 41 |                          | pienet tonttikadut	                         |            |  30 |  30     |  600	   | 141 241 341 441 541 641 |
+| 42 |                          | keskusta, pienet kadut	                     |            |  30 |  23     |  500	   | 142 242 342 442 542 642 |
+
+*) Alun perin 2200, mutta verkolla on käytetty johdonmukaisesti kapasiteettia 2100, Trafixin mukaan siksi,
+että 2200 on epärealistisen suuri kapasiteetti normaaleissa olosuhteissa, ja tekee moottoriteistä liian houkuttelevia vaihtoehtoja.
+
+**) Helmet 3.1:ssä linkin kapasiteettia kasvatettu 1700:sta arvoon 1800 ajon/h (paitsi linkkityypillä x22 ar-voon 1900 ajon/h),
+jos autoliikenteen käytössä on vähintään kolme kaistaa ko. aikajakson aikana.
+
+***) Helmet 3.1:ssä linkin vapaata nopeutta kasvatettu 54:stä arvoon 63 km/h.
```
Comment (johpiip) [inline docs/emme_verkko.md:314]: Do these still apply in Helmet 4.1?

**Review by johpiip — CHANGES_REQUESTED** [review]
Very nice update! I made a few review comments. If you want to, we can discuss them on phone so that I can update them per your instructions, or if you can spare a few minutes, maybe you could clarify them yourself?

**Comment by johpiip** [comment]
> @johpiip, she actually already reviewed it (I am not 100 % sure it was a complete review though), but she did not do it in Github, she did it as a Teams message to me.

Oh okay, I did not know that!

File: docs/emme_verkko.md [hunk 4]
```diff
@@ -0,0 +1,643 @@
...
+| Inkoo	      | 149	            | Myrskylä	  | 504	            | Tammela	  | 834             |
+| Janakkala  	| 165	            | Mäntsälä	  | 505	            | Tuusula	  | 858             |
+| Järvenpää  	| 186	            | Nurmijärvi	| 543	            | Vantaa	  | 92              |
+| Karkkila	  | 224	            | Orimattila	| 560	            | Vihti	    | 927             |
+
+### Sentroidien numerointi
+
+Sijoittelualueet on numeroitu kunnittain seuraavan taulukon mukaisesti.
+
+*Taulukko 6. Sijoittelualueiden numerointi
```
[the remark this answers is not in the document]
Comment (zptro) [reply docs/emme_verkko.md:140]: Yes, this should be fixed.

File: docs/emme_verkko.md [hunk 5]
```diff
@@ -0,0 +1,643 @@
...
+| Sipoo       | 12 000–12 999 | Janakkala   | 21 000–21 499 | Myrskylä    | 29 000–29 499 |
+| Mäntsälä    | 13 000–13 999 | Hattula     | 21 500–21 999 | Lapinjärvi  | 29 500–29 999 |
+| Hyvinkää    | 14 000–14 999 | Loppi       | 22 000–22 499 | Loviisa     | 30 000–30 999 |
+
+## Kulkumuodot ja joukkoliikenteen ajoneuvotyypit
+
+Verkolla käytetyt kulkumuodot (modes) ja joukkoliikenteen ajoneuvotyypit (vehicles):
+- Pääkulkumuodon (auto) h on oltava linkillä sallittu, jos halutaan määrittää esim. kääntymiskieltoja.
+  Verkon koodauksessa kannattaa aina sallia kulkumuoto h.
+  Pääkulkumuoto poistuu automaattisesti linkeiltä sijoittelun yhteydessä:
```
[the remark this answers is not in the document]
Comment (zptro) [reply docs/emme_verkko.md:163]: Good comment!

File: docs/emme_verkko.md [hunk 6]
```diff
@@ -0,0 +1,643 @@
...
+## Linkit
+
+### Linkkien ominaisuudet
+
+*Taulukko 9. Linkkien ominaisuudet*
+
+| kenttä          | autoliikenne                                                 | raideliikenne   | kävely- ja pyöräilylinkit           |
+|-----------------|--------------------------------------------------------------|-----------------|-------------------------------------|
+| type            | ks. taulukko 10                                              | ks. taulukko 10 | ks. taulukko 10                     |
+| length          |                                                              |                 |                                     |
```
[the remark this answers is not in the document]
Comment (zptro) [reply docs/emme_verkko.md:217]: Yes, I did not come up with anything to write other than the obvious, that it represents the length of the link.

File: docs/emme_verkko.md [hunk 7]
```diff
@@ -0,0 +1,643 @@
...
+|	lanes	          | kaistamäärä, bussikaista sisältyy ilmoitettuun kaistamäärään | 1               | 1                                   |
+|	modes         	| hcvkybgde                                                    | tp, m, rj       | af                                  |
```
[the remark this answers is not in the document]
Comment (zptro) [reply docs/emme_verkko.md:219]: Oh, did not think of this. Maybe we could add it to the others as well.

File: docs/emme_verkko.md [hunk 8]
```diff
@@ -0,0 +1,643 @@
...
+|	vdf	            | sijoittelufunktion nro                                       | ei käytössä (0) | ei käytössä (0)                     |
+| ul1	| linkin yhden kaistan kapasiteetti | raitiovaunuverkko: aamu-, päivä- ja iltaliikenteen nopeus, juna- ja metroverkko: ei käytössä (0) | ei käytössä (0) |
+| ul2	            | linkin vapaa nopeus                                          | ei käytössä (0) | ei käytössä (0)                     |
+| ul3           	| autoliikenteen linkeille sijoitellaan aluksi raskas liikenne, ja tulokset tallennetaan linkkiattribuuttiin ul3 (oltava link user datassa eikä extra-attribuutissa, jotta tietoa voidaan käyttää sijoittelufunktioissa)         | ei käytössä (0) | ei käytössä (0)                     |
+| @pyoratieluokka	|                                                              |                 | pyörätien laatuluokka (taulukko 12) |
```
[the remark this answers is not in the document]
Comment (zptro) [reply docs/emme_verkko.md:224]: The car links are used for walking and cycling as well. So maybe they should have the same text as for dedicated walk and bike links. But for rail links we could write "ei käytössä (0)".

File: docs/emme_verkko.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (zptro) [reply docs/emme_verkko.md:228]: Yes, they are written to links. Assignment reads from link attributes.

File: docs/emme_verkko.md (hunk 2 above)
Comment (zptro) [reply docs/emme_verkko.md:251]: Well, this is a matter of taste.

File: docs/emme_verkko.md (hunk 3 above)
Comment (zptro) [reply docs/emme_verkko.md:314]: Yes.

**Comment by zptro** [comment]
> Very nice update! I made a few review comments. If you want to, we can discuss them on phone so that I can update them per your instructions, or if you can spare a few minutes, maybe you could clarify them yourself?

Good comments! However, I do not have the authority to do any changes to this branch. The only way for me right now to do the changes, would be to fork the repo and start a new PR.

**Comment by zptro** [comment]
Fixed according  to the comments and changed accidental tabs into space.

**Review by johpiip — APPROVED** [review]
Looks great to me! I added @eevavesaoja as a reviewer. @zptro, you can either merge this right away, or wait for another round of comments. Thank you for your hard work!

**Comment by zptro** [comment]
We can do revisions as separate PR if needed.

**Review by eevavesaoja — COMMENTED** [review]
Moi! Tässä pari kommenttia/kysymystä sivusta emme_verkko.md (hienoa että tämäkin on koottu githubiin!):

"Sijoitteluskripti muuttaa linkien funktio-, nopeus- ja kapasiteettiattribuutit (vdf, ul1, ul2) type-attribuutin perusteella. Kaikille katu- ja tieverkon linkeille ei ole saatu tuotettua taulukko 11:n mukaista linkkityyppiä. "
Tarkoitetaanko tällä ohjetta, että ”Koodaa yleisesti ottaen pelkkä type”?

Voisi selostaa mitä us2-skripti tekee pääpiirteissään? (jos ei ymmärrä python-skriptiä)

**Comment by johpiip** [comment]
Nämä kais on lisätty uusiksi issueiksi!

> Moi! Tässä pari kommenttia/kysymystä sivusta emme_verkko.md (hienoa että tämäkin on koottu githubiin!):
> 
> "Sijoitteluskripti muuttaa linkien funktio-, nopeus- ja kapasiteettiattribuutit (vdf, ul1, ul2) type-attribuutin perusteella. Kaikille katu- ja tieverkon linkeille ei ole saatu tuotettua taulukko 11:n mukaista linkkityyppiä. " Tarkoitetaanko tällä ohjetta, että ”Koodaa yleisesti ottaen pelkkä type”?
> 
> Voisi selostaa mitä us2-skripti tekee pääpiirteissään? (jos ei ymmärrä python-skriptiä)

**Pull request merged**